frontend/src/container/Sidebar.jsx

```diff
@@ -81,14 +82,15 @@ export class Sidebar extends React.Component {
     if (!this.shouldDisplaySidebar(props)) return
 
     if (
-      props.match.params &&
-      props.match.params.idws &&
-      props.workspaceList.find(ws => ws.isOpenInSidebar) === undefined
+      props.location.pathname.includes('workspaces/') &&
+      !props.workspaceList.some(space => space.isOpenInSidebar === true)
     ) {
-      const workspaceIdInUrl = parseInt(props.match.params.idws)
+      const urlElements = props.location.pathname.split('/')
+      const workspaceIdInUrl = parseInt(urlElements[urlElements.indexOf('workspaces') + 1])
 
-      if (props.workspaceList.find(ws => ws.id === workspaceIdInUrl) !== undefined) {
+      if (props.workspaceList.find(space => space.id === workspaceIdInUrl) !== undefined) {
```

Comment:
since you rename "ws" to "space", you can also rename workspaceIdInUrl

Reply:
Updated.